BUG: Incorrect type hint for creating a recarray with fromrecords

### Describe the issue:

The type stub for `np.rec.fromrecords` has a bug and does not hint that an optional keyword-only argument `formats` has a default value. This has no runtime effect but results in `mypy` complaining unless a redundant `formats=None` is specified in the code.

### Reproduce the code example:

```python
import numpy as np
recarray = np.rec.fromrecords([(1, 2.0), (3, 4.0)], names=['a', 'b'])
```
Running `mypy` on this code gives the following error.

### Error message:

```shell
python/lsst/cell_coadds/numpy_bug.py:2: error: No overload variant of "fromrecords" matches argument types "list[tuple[int, float]]", "list[str]"  [call-overload]
python/lsst/cell_coadds/numpy_bug.py:2: note: Possible overload variants:
python/lsst/cell_coadds/numpy_bug.py:2: note:     def fromrecords(recList: _SupportsArray[dtype[void]] | _NestedSequence[_SupportsArray[dtype[void]]] | tuple[Any, ...] | _NestedSequence[tuple[Any, ...]], dtype: dtype[Any] | type[Any] | _SupportsDType[dtype[Any]] | str | tuple[Any, int] | tuple[Any, SupportsIndex | Sequence[SupportsIndex]] | list[Any] | _DTypeDict | tuple[Any, Any] | None = ..., shape: SupportsIndex | Sequence[SupportsIndex] | None = ..., formats: None = ..., names: None = ..., titles: None = ..., aligned: bool = ..., byteorder: None = ...) -> recarray[Any, dtype[record]]
python/lsst/cell_coadds/numpy_bug.py:2: note:     def fromrecords(recList: _SupportsArray[dtype[void]] | _NestedSequence[_SupportsArray[dtype[void]]] | tuple[Any, ...] | _NestedSequence[tuple[Any, ...]], dtype: None = ..., shape: SupportsIndex | Sequence[SupportsIndex] | None = ..., *, formats: dtype[Any] | type[Any] | _SupportsDType[dtype[Any]] | str | tuple[Any, int] | tuple[Any, SupportsIndex | Sequence[SupportsIndex]] | list[Any] | _DTypeDict | tuple[Any, Any] | None, names: str | Sequence[str] | None = ..., titles: str | Sequence[str] | None = ..., aligned: bool = ..., byteorder: Literal['S', '<', '>', '=', '|', 'L', 'B', 'N', 'I'] | None = ...) -> recarray[Any, dtype[record]]
```


### Python and NumPy Versions:

1.24.4
3.11.7 | packaged by conda-forge | (main, Dec 23 2023, 14:38:07) [Clang 16.0.6 ]


### Runtime Environment:

_No response_

### Context for the issue:

_No response_

--- a/numpy/_core/records.pyi
+++ b/numpy/_core/records.pyi
@@ -174,7 +174,7 @@ def fromrecords(
     dtype: None = ...,
     shape: None | _ShapeLike = ...,
     *,
-    formats: DTypeLike,
+    formats: DTypeLike = ...,
     names: None | str | Sequence[str] = ...,
     titles: None | str | Sequence[str] = ...,
     aligned: bool = ...,

